nix/repl-containers.nix

```diff
@@ -0,0 +1,29 @@
+{ pkgs }:
+let
+  mkImage = pkg: cmd:
+    pkgs.dockerTools.buildLayeredImage {
+      name = "ghcr.io/cofree-coffee/${pkg.name}-repl";
+      created = "now";
+      tag = "latest";
+      contents = [
+        pkg
+      ];
+      config = {
+        Cmd = cmd;
+      };
+    };
+
+  mkPath = xs:
+    with pkgs.lib;
+    with pkgs.lib.attrsets;
+    let paths = mapAttrsToList (name: image: "ln -s ${image} $out/${name}") xs;
+    in "mkdir -p $out" + (foldl' (a: b: a + ";" + b) "" paths);
+
+  images =
+    {
+      python = mkImage pkgs.python3 [ "python" "-iq" ];
+      node = mkImage pkgs.nodejs [ "node" "-i" ];
+      ghci = mkImage pkgs.ghc [ "ghci" ];
+    };
+in
+pkgs.runCommand "repls" { } (mkPath images) 
```

Comment:
@JonathanLorimer for REPL bots I am building docker images with nix. Do you have any thoughts on how we can design this derivation to be more scalable for adding more languages?

Reply:
@JonathanLorimer I don't think that's a good idea. The point of containerization is to isolate the server from untrusted code execution, so that if someone runs `rm -rf /` as a joke it doesn't actually nuke the entire server. You can increase the isolation of the server from these untrusted REPLs by using Linux namespacing stuff, but the more of that you do, the more of existing containerization solutions you reinvent.